lib/testing/src/lib/core/pages/notification-history.page.ts

```diff
@@ -0,0 +1,52 @@
+/*!
+ * @license
+ * Copyright 2019 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { by, element } from 'protractor';
+import { BrowserActions } from '../utils/browser-actions';
+
+export class NotificationHistoryPage {
+
+    notificationList = element(by.css('#adf-notification-history-list'));
+
+    clickNotificationButton() {
+        BrowserActions.clickExecuteScript('#adf-notification-history-open-button');
+    }
+
+    clickAllRead() {
+        BrowserActions.click(element(by.css('#adf-notification-history-mark-as-read')));
+    }
+
+    checkNotificationIsPresent(text: string) {
+        expect(BrowserActions.getText(this.notificationList)).toContain(text);
+    }
+
+    checkNotificationIsNotPresent(text: string) {
```

Comment:
I would suggest having a single method like 
```
async isNotificationPresent(text: string) {
    const list = await BrowserActions.getText(this.notificationList);
    return list.includes(text);
}
```
and have the expect in the actual tests like:
```
expect(notificationHistoryPage.isNotificationPresent('blah blah')).toBe(true, 'some message');
expect(notificationHistoryPage.isNotificationPresent('another blah')).not.toBe(true, 'some message'); // or .toBe(false) ...
```

Reply:
I get your point and it looks nice but the problem here is that after the check I need always to clickMarkAsRead() and close the dialog to be sure it doesn't affect the other test.

Reply:
ah I see :(